cEpiabm/src/simulations/basic_simulation.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "basic_simulation.hpp"
+
+#include "../logfile.hpp"
+
+#include <chrono>
+
+namespace epiabm
+{
+
+    BasicSimulation::BasicSimulation(PopulationPtr population) :
+        m_population(population),
+        m_sweeps(),
+        m_timestepReporters()
+    {
+    }
+
+    void BasicSimulation::addSweep(SweepInterfacePtr sweep)
+    {
+        m_sweeps.push_back(sweep);
+    }
+
+    void BasicSimulation::addTimestepReporter(TimestepReporterInterfacePtr timestepReporter)
+    {
+        m_timestepReporters.push_back(timestepReporter);
+    }
+
+    void BasicSimulation::simulate(const unsigned short timesteps)
+    {
+        auto t0 = std::chrono::system_clock::now();
+        LOG << LOG_LEVEL_NORMAL << "Setting up Simulation.";
+        setup();
+
+        LOG << LOG_LEVEL_NORMAL << "Iterating through timesteps...";
+        //try
+        //{
+            for (const auto& reporter : m_timestepReporters)
+                    reporter->report(m_population, 0);
+
+            for (unsigned short timestep = 1; timestep <= timesteps; timestep++)
+            {
+                // Run Sweeps
+                for (const auto& sweep : m_sweeps) (*sweep)(timestep);
+
+                // Report
+                for (const auto& reporter : m_timestepReporters)
+                    reporter->report(m_population, timestep);
+            }
+        //}
+        // LCOV_EXCL_START
+        /*
+        catch (std::exception& e)
+        {
+            LOG << LOG_LEVEL_ERROR << "Error iterating through timesteps: " << e.what();
+        }*/
+        // LCOV_EXCL_END
```

Comment:
Was this for testing or does it have some function?

Reply:
Undecided. Is better to catch the error so it can be logged. But for testing with a debugger I needed the error to reach the top level and cause a crash.
TLDR: Was for testing

Reply:
Does that mean we don't need it anymore, or is it still used in the automated tests? Just aware that's quite a public facing file so commented out code like that might be confusing!